lib/presentation/view/now_playing_movies_category.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:flutter/material.dart';
+
+import '../../core/utils/enums/endpoints.dart';
+import '../../core/utils/resources/data_state.dart';
+import '../bloc/movies_bloc.dart';
+import '../widgets/movie_list_presentation/list_widgets/container_movie_preview_list.dart';
+
+class NowPlayingMoviesCategory extends StatefulWidget {
+  const NowPlayingMoviesCategory({
+    super.key,
+    required this.moviesBloc,
+  });
+
+  final MoviesBloc moviesBloc;
+
+  @override
+  State<StatefulWidget> createState() => _NowPLayingMoviesState();
+}
+
+class _NowPLayingMoviesState extends State<NowPlayingMoviesCategory> {
+  final Endpoints endpoint = Endpoints.nowPlaying;
+
+  @override
+  void initState() {
+    super.initState();
+    widget.moviesBloc.fetchEndpointsMovies(
+      endpoint,
+    );
+  }
+
+  @override
+  void dispose() {
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        child: StreamBuilder(
+          stream: widget.moviesBloc.nowPlayingMoviesStream,
+          builder: (context, snapshot) {
+            if (snapshot.hasData) {
+              if (snapshot.data is DataFailure) {
+                return Text(snapshot.data!.error.toString());
+              } else if (snapshot.data is DataSuccess) {
+                return ListView.builder(
+                  itemCount: snapshot.data?.data?.length,
+                  itemBuilder: (
+                    BuildContext context,
+                    int index,
+                  ) {
+                    return ContainerMoviePreviewList(
```

Comment:
add some error handling to the itemBuilder callback to prevent the app from crashing if the snapshot.data?.data?[index] is null

Reply:
Done!